### PR TITLE
Use `Scratch` instead of `MutableArtifacts.toml`

### PR DESCRIPTION
**Pull request by staticfloat**
Significantly simplify `gdk-pixbuf` querying by using a scratch space to hold mutable state.
Also try to avoid needing to copy around the pixbuf loaders; instead
opting for a more dynamic approach where we invoke
`gdk-pixbuf-query-loaders` multiple times, concatenating the output
appropriately.

### CODE REVIEW

**Comment by staticfloat** [comment]
@tknopp what do you think about bumping the minimum required version to 1.6?  That's best for `Scratch` usage anyway, since that's the point at which `Pkg.gc()` knows how to clean these things up.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,7 +15,6 @@ jobs:
       fail-fast: false
       matrix:
         julia-version:
-          - "1.3"
           - "1"
```
Comment (giordano) [inline .github/workflows/ci.yml:18]: I'd still run CI on the oldest supported version:
```suggestion
          - "1.6"
          - "1"
```

File: Project.toml [hunk 2]
```diff
@@ -14,6 +14,7 @@ Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
 Librsvg_jll = "925c91fb-5dd6-59dd-8e8c-345e74382d89"
 Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
 Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
+Scratch = "6c6a2e73-6563-6170-7368-637461726353"
```
Comment (giordano) [inline Project.toml:17]: Need to add a compat entry for Scratch.jl

File: Project.toml [hunk 3]
```diff
@@ -33,7 +34,7 @@ Xorg_xkeyboard_config_jll = "2.27.0"
 adwaita_icon_theme_jll = "3.33.92"
 gdk_pixbuf_jll = "2.38.2"
 hicolor_icon_theme_jll = "0.17.0"
-julia = "1.3"
+julia = "1.6"
```
Comment (giordano) [inline Project.toml:37]: Reminder for the maintainers of the package that narrowing the range of supported versions of Julia warrants a minor version bump

**Comment by tknopp** [comment]
Requiring 1.6 is fine for me. Give me a ping once this is ready an passing tests.

**Comment by maleadt** [comment]
+1 on this change; Gtk.jl modifying it's own directory is currently breaking PkgEval for packages that depend on Gtk.jl.

**Pull request merged**